pulp_python/app/models.py

```diff
@@ -92,9 +111,128 @@ class PythonImporter(Importer):
     """
 
     TYPE = 'python'
+    project_names = ArrayField(models.TextField())
 
-    def sync(self):
+    def _fetch_inventory(self):
         """
-        Synchronize the repository with the remote repository.
+        Fetch existing content in the repository.
+        Returns:
+            set: of content filenames.
         """
-        raise NotImplementedError
+        inventory = set()
+
+        q_set = PythonPackageContent.objects.filter(repositories=self.repository)
+        q_set = q_set.only("filename")
+        for content in (c.cast() for c in q_set):
+            inventory.add(content.filename)
+        return inventory
+
+    def _fetch_remote(self):
+        remote = []
+
+        metadata_urls = [urljoin(self.feed_url, 'pypi/%s/json' % project)
```

Comment:
This should be `pypi/%s/json/`. As it is will work for syncing from PyPI, but we need the trailing slash to sync from PyPI and other Pulps.

Reply:
err the trailing slash does not work when accessing warehouse:
https://pypi.org/pypi/pulpcore/json vs https://pypi.org/pypi/pulpcore/json/